docs/contributtion/experimental-code.md

```diff
@@ -0,0 +1,167 @@
+made before an official release.
+
+Experimental code is made available in the following ways:
+
+- Components and exports prefixed with `UNSTABLE_`
+- Feature flags
+- Expanding existing APIs
+
+In all cases, experimental code should be treated as unstable and can change
+between release versions for the package that it is being imported from.
+
+- The API is not fixed, and is likely to change
+- The API is not bound by semver
+- The component, export, feature, etc. may change, be renamed, or removed in the
+  future without warning
+
+## Components and exports prefixed with `UNSTABLE_`
+
+For experimental or unstable exports, we use the `UNSTABLE_` prefix. For
+example:
+
+```js
+// An unstable method
+function UNSTABLE_layout() {
+  // ...
+}
+
+// An unstable variable
+const UNSTABLE_meta = {
+  // ...
+};
+
+// An unstable component will retain its name, specifically for things like
+// the rules of hooks plugin which depend on the correct casing of the name
+function ExampleComponent(props) {
+  // ...
+}
+
+// However, when we export the component we will export it with the `UNSTABLE_`
+// prefix. (Similar to React.unstable_Suspense, React.unstable_Profiler)
+// however we will use it in all caps format
+export { default as UNSTABLE_ExampleComponent } from './components/ExampleComponent';
+```
+
+For teams using these features, they will need to import the functionality by
+using the `UNSTABLE_` prefix. For example:
+
+```jsx
+import { UNSTABLE_ExampleComponent as ExampleComponent } from '@lmc-eu/spirit-web-react';
+```
+
+### Documenting components and exports prefixed with `UNSTABLE_`
+
+Within the experimental section of the `@lmc-eu/spirit-web-react` Storybook, there are
+individual entries for each unstable export. These contain stories and written
+documentation outlining the functionality of these components and exports.
+
+## Feature flags
+
+Some Spirit packages ship with available feature flags. These feature flags
+enable new behavior and styling, allowing you to opt-in to new breaking changes
+while remaining on the current major version. When a new feature flag is
+introduced it is marked `false` or "off" by default to ensure backwards
+compatibility. A feature flag may be configured in JavaScript, SASS, or both.
+
+All the currently available feature flags, as well as how to enable them, is
+documented in the each package main README file in the section `Feature Flags`.
+
+### Feature flag naming convention
+
+All feature flags follow a prefix naming convention that indicate status.
+
+#### Flags prefixed with `enable-*`
+
+- Contain new features that we'd like consuming projects to test
+- Are generally stable and unlikely to change but may change based on user
+  feedback
+- May require some manual migration or code changes within your project
+- Are documented in the main package README file
+- Need user feedback to ensure we've met all concerns relating to this feature
```

Comment:
also, we used it to introduce visual BCs.

Reply:
Added.